app/scripts/modules/core/src/pipeline/config/stages/bakeManifest/bakeManifestStage.ts

```diff
@@ -0,0 +1,23 @@
+import { module } from 'angular';
+
+import { PIPELINE_CONFIG_PROVIDER, PipelineConfigProvider, SETTINGS } from '@spinnaker/core';
+import { BakeManifestConfigCtrl } from './bakeManifestConfig.controller';
+
+export const BAKE_MANIFEST_STAGE = 'spinnaker.core.pipeline.stage.bakeManifestStage';
+
+module(BAKE_MANIFEST_STAGE, [
+  PIPELINE_CONFIG_PROVIDER,
+]).config((pipelineConfigProvider: PipelineConfigProvider) => {
+  if (SETTINGS.feature.versionedProviders) {
+    pipelineConfigProvider.registerStage({
+      label: 'Bake (Manifest)',
+      description: 'Bake a manifest (or multi-doc manifest set) using a template renderer such as Helm.',
+      key: 'bakeManifest',
+      templateUrl: require('./bakeManifestConfig.html'),
+      controller: 'BakeManifestConfigCtrl',
+      producesArtifacts: true,
+      cloudProvider: 'kubernetes',
+      controllerAs: 'ctrl',
+    });
```

Comment:
We'll need to register the `expectedArtifactId` field as an artifact reference so that it can be cleaned up if the template artifact is deleted.

You can see `app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/deployManifestStage.ts` for an example of usage but it'll look approximately like this:

```
artifactReferenceServiceProvider.registerReference('stage', () => [['expectedArtifactId']]);
```